app/tests/test_quizzes.py

```diff
@@ -21,14 +40,70 @@ def test_create_quiz(self):
         response = self.client.get(f"{quizzes.router.prefix}/{quiz_id}")
         assert response.status_code == 200
 
-    def test_get_question_if_id_valid(self):
-        response = self.client.get(f"{quizzes.router.prefix}/{self.id}")
+    def test_get_quiz_if_id_valid(self):
+        response = self.client.get(
+            f"{quizzes.router.prefix}/{self.short_homework_quiz_id}"
+        )
         assert response.status_code == 200
         response = response.json()
-        assert len(response["question_sets"][0]["questions"]) == self.length
+        assert (
+            len(response["question_sets"][0]["questions"])
+            == self.short_homework_quiz_questions_length
+        )
 
     def test_get_quiz_returns_error_if_id_invalid(self):
         response = self.client.get(f"{quizzes.router.prefix}/00")
         assert response.status_code == 404
         response = response.json()
         assert response["detail"] == "quiz 00 not found"
+
+    def test_created_long_quiz_contains_subsets(self):
+        # the created long quiz should contain the same number of questions as the one in the provided input json
+        assert self.long_assessment_quiz_questions_length == len(
+            self.long_assessment_quiz_data["question_sets"][0]["questions"]
```

Comment:
a doubt -- the quizzes will work fine na if there are multiple question sets also? specifying `[0]` is only for test? 

earlier our code catered only to one question set